src/main/java/de/tum/in/www1/artemis/web/rest/QuizExerciseResource.java

```diff
@@ -372,20 +372,20 @@ public QuizExerciseResource(QuizExerciseService quizExerciseService,
         //update QuizExercise
         quizExercise.reconnectJSONIgnoreAttributes();
 
-        //adjust existing results if an answer or and question was deleted and recalculate them
-        quizExerciseService.adjustResultsOnQuizChanges(quizExercise);
+        // QuizExercise result = quizExerciseService.saveWithNoNewEntities(quizExercise);
+        // needed in case the instructor adds a new solution to the question, the quizExercise has to be saved again so that no PersistencyExceptions can appear
+        QuizExercise result = quizExerciseService.save(quizExercise);
```

Comment:
It is very irritating that you call this variable result. I suggest to rename it to updatedQuizExercise or savedQuizExercise

Reply:
I renamed it to **updatedQuizExercise**.